gdnative-sys/src/lib.rs

```diff
@@ -7,46 +7,91 @@
 include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 include!(concat!(env!("OUT_DIR"), "/api_wrapper.rs"));
 
+enum InitError {
+    VersionMismatch {
+        api_type: GDNATIVE_API_TYPES,
+        want: godot_gdnative_api_version,
+        got: godot_gdnative_api_version,
+    },
+    NotFound {
+        message: String,
+    },
+}
+
+unsafe fn report_error(options: *const godot_gdnative_init_options, error: InitError) {
+    use std::ffi::CString;
+    match error {
+        InitError::VersionMismatch {
+            api_type,
+            want,
+            got,
+        } => {
+            if let Some(f) = (*options).report_version_mismatch {
+                f(
+                    (*options).gd_native_library,
+                    CString::new(format!("{}", api_type)).unwrap().as_ptr(),
+                    want,
+                    got,
+                );
+            }
+        }
+        InitError::NotFound { message } => {
+            if let Some(f) = (*options).report_loading_error {
+                f(
+                    (*options).gd_native_library,
+                    CString::new(message).unwrap().as_ptr(),
+                );
+            }
+        }
+    }
+}
```

Comment:
As mentioned in the comment on `from_raw` above, I think this belongs in `-core`.